typescript/sdk/src/token/deploy.ts

```diff
@@ -135,11 +135,30 @@ export class HypERC20Deployer extends GasRouterDeployer<
     chain: ChainName,
     config: HypERC20CollateralConfig,
   ): Promise<HypERC20Collateral> {
-    return this.deployContract(
-      chain,
-      isFastConfig(config) ? TokenType.fastCollateral : TokenType.collateral,
-      [config.token, config.mailbox],
-    );
+    let contractName:
+      | TokenType.fastCollateral
+      | TokenType.collateral
+      | TokenType.collateralVault;
+    let constructorArgs: Parameters<
+      HypERC20Factories[typeof contractName]['deploy']
+    >;
+    switch (config.type) {
+      case TokenType.fastSynthetic || TokenType.fastCollateral:
```

Comment:
why `|| fastSynthetic`?

Reply:
Just reproducing isFastConfig() since its not compatible with switch
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/0338f8111674a643b417c05290350f7237264a7a/typescript/sdk/src/token/config.ts#L73-L75